pkg/registry/flowcontrol/ensurer/flowschema.go

```diff
@@ -18,191 +18,144 @@ package ensurer
 
 import (
 	"context"
-	"errors"
 	"fmt"
 
 	flowcontrolv1beta3 "k8s.io/api/flowcontrol/v1beta3"
 	"k8s.io/apimachinery/pkg/api/equality"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/util/sets"
 	flowcontrolclient "k8s.io/client-go/kubernetes/typed/flowcontrol/v1beta3"
 	flowcontrollisters "k8s.io/client-go/listers/flowcontrol/v1beta3"
 	flowcontrolapisv1beta3 "k8s.io/kubernetes/pkg/apis/flowcontrol/v1beta3"
 )
 
-var (
-	errObjectNotFlowSchema = errors.New("object is not a FlowSchema type")
-)
+// WrapBootstrapFlowSchemas creates a generic representation of the given bootstrap objects bound with their operations
+func WrapBootstrapFlowSchemas(client flowcontrolclient.FlowSchemaInterface, lister flowcontrollisters.FlowSchemaLister, boots []*flowcontrolv1beta3.FlowSchema) BootstrapObjects {
+	return &bootstrapFlowSchemas{
+		flowSchemaClient: flowSchemaClient{
+			client: client,
+			lister: lister},
+		boots: boots,
+	}
+}
 
-// FlowSchemaEnsurer ensures the specified bootstrap configuration objects
-type FlowSchemaEnsurer interface {
-	Ensure([]*flowcontrolv1beta3.FlowSchema) error
+type flowSchemaClient struct {
+	client flowcontrolclient.FlowSchemaInterface
+	lister flowcontrollisters.FlowSchemaLister
 }
 
-// FlowSchemaRemover is the interface that wraps the
-// RemoveAutoUpdateEnabledObjects method.
-//
-// RemoveAutoUpdateEnabledObjects removes a set of bootstrap FlowSchema
-// objects specified via their names. The function removes an object
-// only if automatic update of the spec is enabled for it.
-type FlowSchemaRemover interface {
-	RemoveAutoUpdateEnabledObjects([]string) error
+type bootstrapFlowSchemas struct {
+	flowSchemaClient
```

Comment:
nit: can we make it a named field:
```
client flowSchemaClient
```
?

Reply:
We could but what would be the benefit? The costs would include writing more methods because of promotions that are no longer there.